spec/features/users_features_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.feature "User can sign in and out", type: :feature do
+  context "User not signed in and on the homepage" do
+    it "Should see a 'sign in' link and a 'sign up' link" do
+      visit('/')
+      expect(page).to have_link('Sign in')
+      expect(page).to have_link('Sign up')
+    end
+
+    it "Should not see 'sign out' link" do
+      visit('/')
+      expect(page).not_to have_link('Sign out')
+    end
+  end
+
+  context "User signed in on the homepage" do
+    before do
+      visit('/')
+      click_link('Sign up')
+      fill_in('Email', with: 'test@example.com')
+      fill_in('Password', with: 'testtest')
+      fill_in('Password confirmation', with: 'testtest')
+      click_button('Sign up')
+    end
+
+    it "Should see 'sign out' link" do
```

Comment:
This is not really a feature, generally I save my feature tests for the actual features I want to implement and not worry so much about writing tests for these sorts of things that I would not consider business-critical